File: strategy/java/StrategyTest.java
public class StrategyTest {
    public static void main(String[] args) {
        PaySelector paySelector = new PaySelector();

        Currency curr = Currency.CNY;
        float amount = 500.3F;
        paySelector.getPayStrategy(curr, amount).pay(curr, amount);

        curr = Currency.USD;
        amount = 1500.67F;
        paySelector.getPayStrategy(curr, amount).pay(curr, amount);

        curr = Currency.CNY;
        amount = 2934.33F;
        paySelector.getPayStrategy(curr, amount).pay(curr, amount);

        curr = Currency.USD;
        amount = 5201;
        paySelector.getPayStrategy(curr, amount).pay(curr, amount);

        curr = Currency.CNY;
        amount = 10000;
        paySelector.getPayStrategy(curr, amount).pay(curr, amount);
    }
}

File: strategy/go/strategy_test.go
package strategy

import "testing"

func TestStrategy(t *testing.T) {
	paySelector := PaySelector{}

	var curr, amount = CNY, float32(500.3)
	paySelector.GetPayStrategy(curr, amount).Pay(curr, amount)

	curr, amount = USD, 1500.67
	paySelector.GetPayStrategy(curr, amount).Pay(curr, amount)

	curr, amount = CNY, 2934.33
	paySelector.GetPayStrategy(curr, amount).Pay(curr, amount)

	curr, amount = USD, 5201
	paySelector.GetPayStrategy(curr, amount).Pay(curr, amount)

	curr, amount = CNY, 10000
	paySelector.GetPayStrategy(curr, amount).Pay(curr, amount)
}
